pkg/arvo/sys/arvo.hoon

```diff
@@ -693,26 +694,49 @@
               =^  rey  +>+  (^load +<)
               [rey +>.$]
     ::
-    ++  peek  |=(* (^peek ;;([@da path] +<)))      ::  46
+    ++  peek  |=  *                                     ::  46
+              =/  rob  (^peek ;;([@da path] +<))
+              ?~  rob  ~
+              ?~  u.rob  ~
+              [~ u.u.rob]
```

Comment:
I don't think this is right for the top-level `+peek` arm. At least, it's not obvious to me that the runtime *shouldn't* be able to distinguish between blocking and failing reads. But I see that it's consistent with what was here before. Filing away for later ...

Reply:
Yeah, I agree with that.  Worth revisiting when we start using peek more from the runtime.